### PR TITLE
Update docs, fix typos, etc.

### PR DESCRIPTION
**Pull request by jyeakel**
General review of documentation for grammar, typo, punctuation, etc.

A few other questions came up during the review. @MykolaGolubyev, please weigh in:

1. Internal search is still referenced in the docs. Was this an oversight or was there some reason that we couldn't apply the `firstAvailable` trick with `include-markdown`?
2. The external-facing install steps from Maven Central are still in a TODO state
3. I noticed a lot of repetition between the java.md and java-doc.md files. Same between code-snippets.md and external-code-snippets.md. Is this intentional?

### CODE REVIEW

File: znai-docs/documentation/flow/search.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Avoid: using generic names in your page titles and page section titles. You shou
 
 Global search is performed by [search/](https://search.app.twosigma.com).
 
-Global search consider each page section as an independent entry as well. 
+Global search treates each page section as an independent entry as well.
```
Comment (byc1234) [inline znai-docs/documentation/flow/search.md]: treates to treats?

**Comment by MykolaGolubyev** [comment]
> Internal search is still referenced in the docs. Was this an oversight or was there some reason that we couldn't apply the firstAvailable trick with include-markdown?
We should! Do you know how to do it? Is this something you can help with?

> The external-facing install steps from Maven Central are still in a TODO state
Need to get on to this.

> I noticed a lot of repetition between the java.md and java-doc.md files. Same between code-snippets.md and external-code-snippets.md. Is this intentional?
Could you elaborate? I remember there are repetation between `java` and `external-snippets`. That was done in case users just click on Java without reading external one. Not sure how to arrange it better. 
Maybe just have a page that talks about custom things like `readMore`, `title`, etc. Because it is applicable to most code snippets.

File: znai-docs/documentation/snippets/api-parameters.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ score, Integer, another description line with *markdown* support
 
 # Nested
 
-Use `parent.children` syntax to define nested objects like this
+Use `parent.children` syntax to define nested objects like this.
```
[the remark this answers is not in the document]
Comment (jyeakel) [reply znai-docs/documentation/snippets/api-parameters.md]: I generally just want there to be some punctuation. Whether it's a `.` or `:` is not a science but I usually use the `:` when what immediately follows is a demonstration that is referenced with some direct language ("this", "like so", "as follows"). In this case, I broke that pattern. Will change to `:` :)

File: znai-docs/documentation/flow/search.md (hunk 1 above)
Comment (jyeakel) [reply znai-docs/documentation/flow/search.md]: Yes! Will fix.

File: znai-docs/documentation/layout/jupyter-notebook-two-sides.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ Note: Content below is included using `include-jupyter`
 
 # Story First
 
-Use `storyFirst` if you use a notebook with Markdown cells to tell a story. In this mode code will be moved to the right
+Use `storyFirst` if you use a notebook with Markdown cells to tell a story. In this mode, code will be moved to the right
```
[the remark this answers is not in the document]
Comment (jyeakel) [reply znai-docs/documentation/layout/jupyter-notebook-two-sides.md]: Should it be added?

**Comment by jyeakel** [comment]
> We should! Do you know how to do it? Is this something you can help with?

I can give it a try.

> Could you elaborate? I remember there are repetation between java and external-snippets. That was done in case users just click on Java without reading external one. Not sure how to arrange it better.
> Maybe just have a page that talks about custom things like readMore, title, etc. Because it is applicable to most code snippets.

The `java.md` file has several sections related to JavaDoc that seem mostly to cover the same material in`java-doc.md`. And for the code-snippets pages, yeah it just goes over the same material with regard to wide code, highlights, titles, etc. I get if its because we want to make sure people see it if they land on one page and not the other, but maybe we should be reusing markdown at least.

**Comment by MykolaGolubyev** [comment]
> The java.md file has several sections related to JavaDoc that seem mostly to cover the same material injava-doc.md. 

That's so weird. I thought when I moved stuff to `java-doc.md` I removed the old one. Maybe some merge issue. Please remove it while you are there. 

> And for the code-snippets pages, yeah it just goes over the same material with regard to wide code, highlights, titles, etc. I get if its because we want to make sure people see it if they land on one page and not the other, but maybe we should be reusing markdown at least.

Reusing markdown could be tricky as plugin name and file names used in examples are different. Maybe I could cook something up by using templates.

**Comment by jyeakel** [comment]
Okay, made some fixes and deleted the extra JavaDoc stuff. I'll handle the search stuff in a separate PR since it will also require some work internally to set up the alternative Markdown.

**Comment by MykolaGolubyev** [comment]
Go for merge!

**Pull request merged**